src/hooks/useFetch.js

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from 'react';
+
+const fetchUrl = (url, after, count, limit = 100, time = 'year') => {
+  const urlParams = `t=${time}&limit=${limit}${
+    count && after ? `&after=${after}&count${count}` : ''
+  }`;
+  return `${url}/top.json?${urlParams}`;
+};
+
+const makeRequest = async (url) => {
+  const res = await fetch(url);
+
+  if (!res.ok) {
+    throw new Error('Unable to fetch data');
+  }
+
+  const data = await res.json();
+  return data;
+};
+
+const fetchPosts = async (
+  url,
+  posts = [],
+  count = null,
+  after = null,
+  numRequests = 500,
+) => {
+  if (count >= numRequests) {
+    return posts;
+  }
+
+  const currentUrl = fetchUrl(url, after, count);
+
+  const data = await makeRequest(currentUrl);
+
+  const recordsRetrieved = data.data.dist;
```

Comment:
**Naming:** This sounds like these are actual records (= posts) but it's a number